Backend/CourseService.cs

```diff
@@ -49,9 +54,9 @@ public Task<List<Course>> FindByStartName(string courseStartsWith)
         /// If course with "courseName" exists function return ID, if course doesnt exist fuction
         /// save this course and return id of saved course.
         /// </summary>
-        public async Task<Guid> GetOrAddNotExistsCourseId(string courseName, Block courseBlock)
+        public async Task<Guid> GetOrAddNotExistsCourseId(string courseName, Block courseBlock, bool byName)
```

Comment:
so parameter courseName is also course shortcut:-(, instead of bool byName use two attributes: courseName and course shortcut...
O better, create two methods.. GetOrAddNotExistsCourseIdByName and GetCourseByShortCut...